src/citrine/resources/branch.py

```diff
@@ -61,6 +61,59 @@ def experiment_datasource(self) -> Optional[ExperimentDataSource]:
         branch_erds_iter = erds.list(branch_id=self.uid, version="latest")
         return next(branch_erds_iter, None)
 
+    def update_data(self,
+                    *,
+                    use_existing=True,
+                    retrain_models=False):
+        """
+        Automatically advance the branch to the next version.
+
+        If there are no newer versions of data sources used by this branch this method returns
+        without doing anything
+
+        Parameters
+        ----------
+        use_existing: bool
+            If true the workflows in this branch will use existing predictors that are using
+            the latest versions of the data sources and are ready to use.
+
+        retrain_models: bool
+            If true, when new versions of models are created, they are automatically
+            scheduled for training.
+
+        Returns
+        -------
+        Branch
+            The new branch record after version update
+
+        """
+        path_template = (f'/projects/{{project_id}}/branches/'
+                         f'{{branch_id}}/data-version-updates-predictor')
+        path = format_escaped_url(path_template, project_id=self.project_id, branch_id=self.uid)
+        data = self.session.get_resource(path, version="v2")
+        version_updates = BranchDataUpdate.build(data)
+        # If no new data sources, then exit, nothing to do
+        if len(version_updates.data_updates) == 0:
+            return self
+
+        path_template = f'/projects/{{project_id}}/branches/next-version-predictor'
```

Comment:
Similar to above: `path_template = f'{self._path_template}/next-version-predictor'`